database_sanitizer/utils/mysql.py

```diff
@@ -0,0 +1,167 @@
+
+    if len(url.path) < 2 or not url.path.startswith("/"):
+        raise ValueError("Name of the database is missing from the URL")
+
+    args.append(url.path[1:])
+
+    return args, env
+
+
+MYSQL_NULL_PATTERN = re.compile(r"^NULL$", re.IGNORECASE)
+MYSQL_BOOLEAN_PATTERN = re.compile(r"^(TRUE|FALSE)$", re.IGNORECASE)
+MYSQL_FLOAT_PATTERN = re.compile(r"^[+-]?\d*\.\d+([eE][+-]?\d+)?$")
+MYSQL_INT_PATTERN = re.compile(r"^\d+$")
+MYSQL_STRING_PATTERN = re.compile(r"'(?:[^']|''|\\')*(?<![\\])'")
+
+
+def decode_mysql_literal(text):
+    """
+    Attempts to decode given MySQL literal into Python value.
+
+    :param text: Value to be decoded, as MySQL literal.
+    :type text: str
+
+    :return: Python version of the given MySQL literal.
+    :rtype: any
+    """
+    if MYSQL_NULL_PATTERN.match(text):
+        return None
+
+    if MYSQL_BOOLEAN_PATTERN.match(text):
+        return text.lower() == "true"
+
+    if MYSQL_FLOAT_PATTERN.match(text):
+        return float(text)
+
+    if MYSQL_INT_PATTERN.match(text):
+        return int(text)
+
+    if MYSQL_STRING_PATTERN.match(text):
+        return decode_mysql_string_literal(text)
+
+    raise ValueError("Unable to decode given value: %r" % (text,))
+
+
+MYSQL_STRING_ESCAPE_SEQUENCE_PATTERN = re.compile(r"\\(.)")
+MYSQL_STRING_ESCAPE_SEQUENCE_MAPPING = {
+    "\\0": "\000",
+    "\\b": "\b",
+    "\\n": "\n",
+    "\\r": "\r",
+    "\\t": "\t",
+    "\\Z": "\032",
+}
+
+
+def decode_mysql_string_literal(text):
+    """
+    Removes quotes and decodes escape sequences from given MySQL string literal
+    returning the result.
+
+    :param text: MySQL string literal, with the quotes still included.
+    :type text: str
+
+    :return: Given string literal with quotes removed and escape sequences
+             decoded.
+    :rtype: str
+    """
+    assert text.startswith("'")
+    assert text.endswith("'")
+
+    # Ditch quotes from the string literal.
+    text = text[1:-1]
+
+    return MYSQL_STRING_ESCAPE_SEQUENCE_PATTERN.sub(
+        unescape_single_character,
+        text,
+    )
+
+
+def unescape_single_character(match):
+    """
+    Unescape a single escape sequence found from a MySQL string literal,
+    according to the rules defined at:
+    https://dev.mysql.com/doc/refman/5.6/en/string-literals.html#character-escape-sequences
+
+    :param match: Regular expression match object.
+
+    :return:
+    :rtype: str
+    """
+    value = match.group(0)
+    assert value.startswith("\\")
+    return MYSQL_STRING_ESCAPE_SEQUENCE_MAPPING.get(value) or value[1:]
+
+
+def encode_mysql_literal(value):
+    """
+    Converts given Python value into MySQL literal, suitable to be used inside
+    `INSERT INTO` statement.
+
+    :param value: Value to convert into MySQL literal.
+    :type value: any
+
+    :return: Given value encoded into MySQL literal.
+    :rtype: str
+    """
+    if not pymysql:
+        raise RuntimeError("You need to install PyMySQL for MySQL support")
```

Comment:
Why are you delaying this error to runtime? Wouldn't it be better to report the error as soon as possible, i.e. import time?

It would be no fun, if hitting this error after already running a sanitizer for an hour and then comes the first table which has a configured sanitizer and this function gets called first time with this error. :-P

My suggestion: Either just remove the try-except from the pymysql import (and this check) or move this `raise RuntimeError` to the except block in line 11.

Reply:
I thought it was a good idea to be able to `import` files from the package without having all optional dependencies installed, but it was indeed a bad idea.